include/RAJA/policy/desul/atomic_desul_openmp.hpp

```diff
@@ -0,0 +1,194 @@
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicInc(omp_atomic, T volatile *acc, T val)
+{
+  return desul::atomic_fetch_add(const_cast<T*>(acc)
+                                , val
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicDec(omp_atomic, T volatile *acc)
+{
+  return desul::atomic_fetch_dec(const_cast<T*>(acc)
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicDec(omp_atomic, T volatile *acc, T val)
+{
+  return desul::atomic_fetch_sub(const_cast<T*>(acc)
+                                , val
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicAnd(omp_atomic, T volatile *acc, T value)
+{
+  return desul::atomic_fetch_and(const_cast<T*>(acc)
+                                , value
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicOr(omp_atomic, T volatile *acc, T value)
+{
+  return desul::atomic_fetch_or(const_cast<T*>(acc)
+                                , value
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicXor(omp_atomic, T volatile *acc, T value)
+{
+  return desul::atomic_fetch_xor(const_cast<T*>(acc)
+                                , value
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+}
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicExchange(omp_atomic, T volatile *acc, T value)
+{
+  (void) acc;
+  (void) value;
+  return T{0};
+  /*
+  return desul::atomic_exchange(const_cast<T*>(acc)
+                                , value
+                                , raja_default_desul_order{}
+                                , raja_default_desul_scope{});
+  */
+}
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T atomicCAS(omp_atomic, T volatile *acc, T compare, T value)
+{
+  return desul::atomic_compare_exchange_strong(const_cast<T*>(acc)
+                                , compare
+                                , value
+                                , raja_default_desul_order{} // success order
+                                , raja_default_desul_order{} // failure order
+                                , raja_default_desul_scope{});
```

Comment:
@trws Just wanted to double check that the success and failure memory orders here make sense.

Reply:
Yeah, that's perfectly fine.  If we had a stronger memory order by default, it might help on a small subset of platforms (powerpc for example) to have a weaker one on failure, but it's more consistent to keep it the same until we have a way for the user to decide for themselves.